spec/systems/search_keywords_spec.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'search keywords', type: :system do
+  ActiveJob::Base.queue_adapter = :test
+
+  context 'given exactly match search keyword' do
+    it 'displays searched keywords' do
+      user = Fabricate(:user)
+      keyword = Fabricate(:keyword, user_id: user[:id], keyword: 'AWS')
+
+      visit keywords_path
+
+      within 'form' do
+        fill_in('username', with: user[:username])
+        fill_in('password', with: 'password')
+
+        click_button(I18n.t('auth.login'))
+      end
+
+      expect(current_path).to eql(keywords_path)
+      expect(page).to have_selector('table')
+
+      VCR.use_cassette('with_top_position_adwords', record: :none) do
+        GoogleScrapingJob.perform_now(keyword.id, keyword.keyword)
+      end
+
+      visit keywords_path
+
+      within '#searchForm' do
+        fill_in('search', with: 'AWS')
+
+        click_button(I18n.t('keyword.search'))
+      end
+
+      within 'table' do
+        within 'tbody' do
+          expect(page).to have_selector('tr', count: 1)
+
+          tr_list = all('tr')
+
+          expect(tr_list[0]).to have_selector('td', text: 'AWS')
+        end
+      end
+    end
+  end
+
+  context 'given partially match search keyword' do
+    it 'displays searched keywords' do
+      user = Fabricate(:user)
+      keyword = Fabricate(:keyword, user_id: user[:id], keyword: 'AWS')
+
+      visit keywords_path
+
+      within 'form' do
+        fill_in('username', with: user[:username])
+        fill_in('password', with: 'password')
+
+        click_button(I18n.t('auth.login'))
+      end
+
+      expect(current_path).to eql(keywords_path)
+      expect(page).to have_selector('table')
+
+      VCR.use_cassette('with_top_position_adwords', record: :none) do
+        GoogleScrapingJob.perform_now(keyword.id, keyword.keyword)
+      end
+
+      visit keywords_path
+
+      within '#searchForm' do
+        fill_in('search', with: 'AW')
+
+        click_button(I18n.t('keyword.search'))
+      end
+
+      within 'table' do
+        within 'tbody' do
+          expect(page).to have_selector('tr', count: 1)
+
+          tr_list = all('tr')
+
+          expect(tr_list[0]).to have_selector('td', text: 'AWS')
+        end
+      end
+    end
+  end
+
+  context 'given unmatch search keyword' do
+    it 'displays searched keywords' do
+      user = Fabricate(:user)
+      keyword = Fabricate(:keyword, user_id: user[:id], keyword: 'AWS')
+
+      visit keywords_path
+
+      within 'form' do
+        fill_in('username', with: user[:username])
+        fill_in('password', with: 'password')
+
+        click_button(I18n.t('auth.login'))
+      end
+
+      expect(current_path).to eql(keywords_path)
+      expect(page).to have_selector('table')
+
+      VCR.use_cassette('with_top_position_adwords', record: :none) do
+        GoogleScrapingJob.perform_now(keyword.id, keyword.keyword)
```

Comment:
Do we still need this as the keyword is created in the database already and we removed the upload step? check for other test file too @gutakk

Reply:
Good catch. We don't need this because `search` feature is not related to scraping.
Fixed in 30d5db46ac6df00c4a2a62078cc0c86a2adb8661